src/MIDebugEngine/Engine.Impl/Variables.cs

```diff
@@ -90,6 +90,23 @@ internal sealed class VariableInformation : IVariableInformation
         private string DisplayHint { get; set; }
         public bool IsPreformatted { get; set; }
 
+        public string Address()
+        {
+            // ask GDB to evaluate "&expression"
+            string command = "&("+_strippedName+")";
+            IVariableInformation execVariable = new VariableInformation(command, this);
+            execVariable.SyncEval();
+            return _engine.DebuggedProcess.Natvis.FormatDisplayString(execVariable);
```

Comment:
I think you want to use MI Commands directly instead of going through the VariableInformation classes. The variable information classes do extra work, like applying natvis, setting the output radix that you don't want, as that does extra work, and also may reformat the pointer in a way that might break things.

Looking at [MI documentation](https://sourceware.org/gdb/onlinedocs/gdb/GDB_002fMI-Data-Manipulation.html#GDB_002fMI-Data-Manipulation) I think you can use the `-data-evaluate-expression` command instead of the VarCreate stuff to avoid creating variables for no reason.

Reply:
Changed to use EvalDependentExpression()